### PR TITLE
Add LUCI config file for plugins

### PR DESCRIPTION
**Pull request by keyonghan**
This is part of https://github.com/flutter/flutter/issues/56143#issuecomment-685843581, adding configs for windows try builder.

### CODE REVIEW

File: config/plugins_config.star [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+"""
+Configurations for the plugins repository.
+"""
+
+load("//lib/common.star", "common")
+load("//lib/repos.star", "repos")
+
+def _setup():
```
Comment (godofredoc) [inline config/plugins_config.star:12]: Docs?

File: config/plugins_config.star [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    platform_args = {
+        "windows": {
+            "caches": [swarming.cache(name = "pub_cache", path = ".pub-cache")],
+        },
+    }
+    plugins_define_recipes()
+    plugins_try_config(platform_args)
+
+def plugins_define_recipes():
+    # Defines recipes
```
Comment (godofredoc) [inline config/plugins_config.star:22]: Add this line as doc and add more info?

File: config/plugins_config.star [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    luci.recipe(
+        name = "plugins/plugins",
+        cipd_package = "flutter/recipe_bundles/flutter.googlesource.com/recipes",
+        cipd_version = "refs/heads/master",
+    )
+
+def plugins_try_config(platform_args):
+    # Defines a list view for try builders
```
Comment (godofredoc) [inline config/plugins_config.star:30]: Same as above, please update as a doc and document the argument with an example.

File: config/plugins_config.star [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    # Defines plugins Windows platform try builders
+    common.windows_try_builder(
+        name = "Windows Plugins|windows",
+        recipe = "plugins/plugins",
+        list_view_name = list_view_name,
+        repo = repos.PLUGINS,
+        add_cq = True,
+        caches = [
+            swarming.cache(name = "pub_cache", path = ".pub_cache"),
+        ],
```
Comment (godofredoc) [inline config/plugins_config.star:46]: Use platform args here or remove the parameter and definition at the top?

File: config/plugins_config.star [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        title = "Plugins try builders",
+    )
+
+    # Defines plugins Windows platform try builders
+    common.windows_try_builder(
+        name = "Windows Plugins|windows",
+        recipe = "plugins/plugins",
+        list_view_name = list_view_name,
+        repo = repos.PLUGINS,
+        add_cq = True,
```
Comment (godofredoc) [inline config/plugins_config.star:43]: Do we want this builder to block submissions in the recipes repo?

Comment (keyonghan) [reply config/plugins_config.star:43]: What is the context about add_cq property, or any doc? I see builders from all other repos have this property set as `True`.

File: config/plugins_config.star (hunk 4 above)
Comment (keyonghan) [reply config/plugins_config.star:46]: Changed to use args.

File: config/plugins_config.star (hunk 1 above)
Comment (keyonghan) [reply config/plugins_config.star:12]: Added.

File: config/plugins_config.star (hunk 3 above)
Comment (keyonghan) [reply config/plugins_config.star:30]: Done.

File: config/plugins_config.star (hunk 5 above)
Comment (keyonghan) [reply config/plugins_config.star:43]: Removing it for now.

Comment (godofredoc) [reply config/plugins_config.star]: This is used to decide whether or not to add the builder to the commit_queue.cfg file. commit_queue.cfg is used to decide which builders will run on when a recipe is affected. E.g. if there is a change to the plugins/plugins recipe and add_cq=True then this builder will be run on that recipe change. Setting this value to True is the correct path to follow once we validate the builder runs correcty.

File: config/plugins_config.star [hunk 6]
```diff
@@ -18,14 +19,23 @@ def _setup():
...
 
+# Detailed builder configures for different platforms.
+#
+# [platform_args] has map structure, with platforms as keys.
+# Example:  
+# {
+#    "windows": {
+#        "caches": [swarming.cache(name = "pub_cache", path = ".pub-cache")],
+#    }
+# }
```
Comment (godofredoc) [inline config/plugins_config.star:38]: rm if not used.

Comment (keyonghan) [reply config/plugins_config.star:38]: We are using `platform_args` now (`**platform_args["windows"]`).

Comment (godofredoc) [reply config/plugins_config.star:38]: I see, then if this is the doc for the function please move it inside the function.

File: config/plugins_config.star (hunk 1 above)
Comment (keyonghan) [reply config/plugins_config.star:12]: Ahh, okay. Updated all docs to python format.

File: config/plugins_config.star (hunk 6 above)
Comment (keyonghan) [reply config/plugins_config.star]: Done.

File: config/plugins_config.star [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        cipd_package = "flutter/recipe_bundles/flutter.googlesource.com/recipes",
+        cipd_version = "refs/heads/master",
+    )
+
+def plugins_try_config(platform_args):
+    """Detailed builder configures for different platforms.
+    
+    Args:
+        platform_args (dict) - platform arguments.
+        For example:
```
Comment (godofredoc) [inline config/plugins_config.star:35]: platform_args: The platform arguments passed to luci builder ...
   line continuation is indented to not confuse with other arguments.

Comment (keyonghan) [reply config/plugins_config.star]: Done. We don't need to specify type here?

Comment (keyonghan) [reply config/plugins_config.star]: I suppose the type as missed in the comment. Just added it back.

**Review by godofredoc — APPROVED** [review]

**Pull request merged**